changelog/dmd.source-archive.dd

```diff
@@ -0,0 +1,107 @@
+# Support Source Archive Files
+
+This is a compiler feature, not a D language feature.
```

Comment:
Does such a thing exist? There's basically two poles here in that one the one hand you have "feature works with dmd, not ldc; breaks ldc i.e. D used in industry" or "feature works with dmd, not ldc; no one uses feature"

Having more than one way of doing the same thing has to be really profitable to justify the complexity, I'm not sold this does.

Reply:
It would be up to gdc and ldc if they want to incorporate this or not.

Reply:
First you have to define "profitable". While you may not see it, others do and to them it will be profitable. If I stripped down the D language to only what I _personally_ profitable, D would be a _very_ different language that what it currently is.

Reply:
This isn't stripping anything down, it adds complexity in exchange for not much of a win:

`strace -c -e ... dmd ...` my most recent compiler invocation that about 0.28551851851% (10ms or so) of the total time is spent waiting for fs related system calls

- 31% of that time is `stat` looking for files <- this is where the real speedup usually is (more precise lookup, the actual read is cheap).
- Most of the files its looking for are from dub which probably won't buy into this scheme.

Reply:
As I said here: https://github.com/dlang/dmd/pull/16348#issuecomment-2030668740, the primary benefit is *not* seek performance.